src/client/lazy-app/Compress/index.tsx

```diff
@@ -126,13 +130,18 @@ async function preprocessImage(
 ): Promise<ImageData> {
   assertSignal(signal);
   let processedData = data;
+  const { rotate, flip, crop } = preprocessorState;
 
-  if (preprocessorState.rotate.rotate !== 0) {
-    processedData = await workerBridge.rotate(
-      signal,
-      processedData,
-      preprocessorState.rotate,
-    );
+  if (flip && (flip.horizontal || flip.vertical)) {
+    processedData = await workerBridge.flip(signal, processedData, flip);
+  }
+
+  if (rotate.rotate !== 0) {
+    processedData = await workerBridge.rotate(signal, processedData, rotate);
+  }
+
+  if (crop && (crop.left || crop.top || crop.right || crop.bottom)) {
```

Comment:
```suggestion
  if (crop.left || crop.top || crop.right || crop.bottom) {
```

Don't think the truthy check is needed. If it is, something is wrong with the types.